pandas/io/excel.py

```diff
@@ -66,27 +69,211 @@ def get_writer(engine_name):
         raise ValueError("No Excel writer '%s'" % engine_name)
 
 
+class BaseFile(object):
+    """Base class for excel readers
+
+    A file class can be initialized even if the engine is not installed.
+    If the engine is not installed, io_class and _open_workbook are None.
+    When attempting to use open_workbook while _open_workbook is None, the
+    relevant ImportError is raised. When calling is_ext, False is retunred when
+    the reader is not installed.
+    """
+    # added in order to deal with both Open Document Format and MS Excel
+    # fil types, see PR #9070
+
+    def __init__(self, engine=None):
+        """Set the engine name, and extension. If the engine is not installed,
+        io_class and _open_workbook are both None, but the ImportError is
+        ingored at the init stage. This is done in order to be able to fail
+        only when attempting to open a file of a supported but not installed
+        excel reader. We still need to be able to initialize the supported
+        file reader class that inherits from BaseFile, withouth failing, in
+        order to be able to deduce which file type based on a path extension.
+        """
+        self.engine = engine
+        self.io_class = None
+        self._open_workbook = None
+        # fail silently if engine is not presented on init, only raise when
+        # attempting to use it when opening a workbook
+        try:
+            self.load_engine()
+        except ImportError:
+            pass
+#        if io is not None:
+#            self.engine, self.book = self.create(io, engine=engine)
+#        return self.engine, self.book
+
+    def open_workbook(self, *args, **kwargs):
+        """Explicitely load the engine again (and trigger an ImportError in the
+        process) if _open_workbook is set to None.
+        """
+        # try to load the engine again and raise import error if required
+        if self._open_workbook is None:
+            self.load_engine()
+        # just in case the user passes an already opened workbook of io_class
+        if len(args) > 0 and isinstance(args[0], self.io_class):
+            return args[0]
+        return self._open_workbook(*args, **kwargs)
+
+    def is_ext(self, path):
+        """Verify if the path's extension is supported by the reader. If the
+        reader is not installed, return False
+        """
+        ext = path.split('.')[-1]
+        if self.io_class is None:
+            return False
+        return ext in self.extensions
+
+    def load_engine(self):
+        """Load the engine if installed, but raise ImportError otherwise.
+        If engine is not defined, do nothing.
+        """
+        try:
```

Comment:
I don't think you need this double exception here. the ``self._load_engine()`` should simply raise the appropriate exception.